tests/ui/consts/promoted_const_call.rs

```diff
@@ -1,6 +1,6 @@
 //@ known-bug: #103507
```

Comment:
This bug is long-closed. Something is fishy here.

Reply:
I bet it was a typo lol

Reply:
This is not a known bug, but actually just a check-fail test: #105085

I'll remvoe the known-bug.

Reply:
That's concerning, how did that get added in the first place... this was in https://github.com/rust-lang/rust/pull/114134.

Reply:
We should probably check the other tests that reference that issue
```
tests/ui/consts/issue-94675.rs
1://@ known-bug: #103507

tests/ui/consts/const-block-const-bound.rs
1://@ known-bug: #103507

tests/ui/consts/promoted_const_call.rs
1://@ known-bug: #103507

tests/ui/impl-trait/normalize-tait-in-const.rs
1://@ known-bug: #103507
```
(not in this PR)

Reply:
Just a sweeping change where almost everything const traits related was known bug as the impl was ripped out